Update to Firefox version 140

tschneidereit opened the pull request:
This includes substantial changes to the build setup, including moving all of the dependency installation out of the shell script and into the CI workflow definition.

Before, we were using the Firefox build system's support for installing toolchains. With this change, we instead install them manually. This means we're not tied to Mozilla's handling of toolchain artifacts, which is explicitly documented as only being supported for current versions of the Firefox codebase.

In combination, these changes lead to faster and easier builds, and make it much easier to do local builds, too.

tschneidereit committed:
Switch to new repository, forked off of mozilla-firefox/firefox

tschneidereit committed:
Update to Firefox version 140

This includes substantial changes to the build setup, including moving all of the dependency installation out of the shell script and into the CI workflow definition.

Before, we were using the Firefox build system's support for installing toolchains. With this change, we instead install them manually. This means we're not tied to Mozilla's handling of toolchain artifacts, which is explicitly documented as only being supported for current versions of the Firefox codebase.

In combination, these changes lead to faster and easier builds, and make it much easier to do local builds, too.

tschneidereit committed:
Update `actions/cache` to v4

tschneidereit committed:
parallelize debug and release builds

JakeChampion commented on gecko-revision:
```diff
@@ -1 +1 @@
-7acfe5fcdb611518ecc36f5720cb784fdda1ad08
+wasi-ff140
```
just to note that this is a branch name and not a commit - which means that the build could end up using a different version of bytecodealliance/firefox if someone pushes to this branch

tschneidereit replied:
it's not: it's a tag name. Which still has the same issue in principle, but it's certainly not customary to update tags. And I like how much more explicit this makes things

JakeChampion replied:
It looks to be a branch and not a tag

https://github.com/bytecodealliance/firefox/tree/wasi-ff140

![Screenshot_2025-07-29-21-25-31-36_d0ef7a9810af20f7b394f4f927b723b0.jpg](https://github.com/user-attachments/assets/f87ddd8b-15c8-4620-ad70-c4564d5fa81c)

I couldn't see a tag for it under https://github.com/bytecodealliance/firefox/tags

tschneidereit replied:
Oh, right you are—my apologies! My intent was to use a tag, but clearly I had a different opinion at some point, or made a mistake. Good thing this hasn't landed yet, then.

Additionally, the tag should probably be directly derived from one of the tags on the commit this is all based on. Something like `FIREFOX_140_0_4_RELEASE_STARLING`

JakeChampion replied:
yes, that'd be really nice :D

tschneidereit replied:
I changed things to using that exact tag name now. Thank you for raising this! <3

tschneidereit committed:
Address GH Actions deprecation warning for `set-output`

tschneidereit committed:
Add `js-confdefs.h` to the release bundle's `include` dir

This contains more definitions than the `js-config.h` header which is already included, including vital ones determining the object layout for certain objects, and function signatures. Without these definitions, embeddings create subtly wrong results in very hard to debug ways, I can report from experience.

tschneidereit committed:
Use tag for Firefox revision to pull
